core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -213,13 +213,21 @@ public Object run() throws Exception {
         } catch (PrivilegedActionException e) {
             if (e.getCause() != null && e.getCause().getCause() != null && e.getCause().getCause() instanceof
                     UserStoreException) {
-                if (((UserStoreException) e.getCause().getCause()).getErrorCode() != null) {
-                    throw new UserStoreException(e.getCause().getCause().getMessage(),
-                            ((UserStoreException) e.getCause().getCause()).getErrorCode(), e);
+                boolean isClientException = e.getCause().getCause() instanceof UserStoreClientException;
+                String errorCode = ((UserStoreException) e.getCause().getCause()).getErrorCode();
+                String errorMessage = e.getCause().getCause().getMessage();
+                // If the error codes are available send the error codes as well.
+                if (StringUtils.isNotBlank(errorCode)) {
```

Comment:
Arrage as follows for more readability
if (isClientException) {
     if (StringUtils.isNotBlank(errorCode)) {
      }
  
} else {

}

Reply:
Fixed in 9abb9f1bd36da933af860bd929be65ef1d9f23dd